src/api/callApiWithErrorHandling.ts

```diff
@@ -0,0 +1,34 @@
+import { PathsWithMethod } from 'openapi-typescript-helpers';
+
+import { OperationResponse } from '@stacks/blockchain-api-client';
+import { paths } from '@stacks/blockchain-api-client/lib/generated/schema';
+
+import { logError } from '../common/utils/error-utils';
+import { getErrorMessage } from './getErrorMessage';
+import { useApiClient } from './useApiClient';
+
+const ERROR_TRANSACTION_NAME = 'api-call-error';
+
+type ExtractPath<Endpoint extends keyof paths> = paths[Endpoint];
+
+type ApiParams<Endpoint extends keyof paths> = ExtractPath<Endpoint> extends {
+  get: { parameters: infer Params };
+}
+  ? { params: Params }
+  : { params?: never };
+
+export async function callApiWithErrorHandling<Endpoint extends PathsWithMethod<paths, 'get'>>(
+  apiClient: ReturnType<typeof useApiClient>,
+  apiUrl: Endpoint,
+  apiParams?: ApiParams<Endpoint>
+): Promise<OperationResponse[Endpoint]> {
+  const { error, data } = await apiClient.GET(apiUrl, apiParams as any);
```

Comment:
This function acts as a wrapper around the API client, it enforce type safety for the endpoint, parameters, and response, as well as providing centralized error handling. I had to use any when passing apiParams because the types provided by client library are not fully generic and rely on rigid definitions that cannot be dynamically inferred. Open to suggestions here.

Reply:
I think this works. We'll do with the missing types for now, unless it hinders us in some signifcant way. Would be interesting to see if we could have done this with middleware, but I think that is a separate investigation. Thanks for making these changes!